packages/vite/src/node/config.ts

```diff
@@ -538,21 +542,31 @@ export async function resolveConfig(
   const createResolver: ResolvedConfig['createResolver'] = (options) => {
     let aliasContainer: PluginContainer | undefined
     let resolverContainer: PluginContainer | undefined
+
+    const plugins =
+      options?.plugins?.filter(
+        (p) => p.resolveId && p.name !== 'alias' && !p.name.startsWith('vite:')
+      ) || []
+
     return async (id, importer, aliasOnly, ssr) => {
       let container: PluginContainer
       if (aliasOnly) {
         container =
           aliasContainer ||
           (aliasContainer = await createPluginContainer({
             ...resolved,
-            plugins: [aliasPlugin({ entries: resolved.resolve.alias })]
+            plugins: [
+              ...plugins,
+              aliasPlugin({ entries: resolved.resolve.alias })
```

Comment:
Shouldn't this be inverted? The alias plugin is the first one in the regular pipeline.